OneDrive-Cloud-Player/Services/Helpers/CacheHelper.cs

```diff
@@ -546,5 +555,30 @@ public CachedDriveItem GetParentItemByParentItemId(CachedDrive SelectedDriveFold
             }
         }
 
+        /// <summary>
+        /// Filter a list of drive items for playable media.
+        /// </summary>
+        /// <param name="unfilteredList">The list to filter</param>
+        /// <returns>A list of CachedDriveItems with only playable files, an empty list in case
+        /// the input contains unplayable files or null in case the input is null.</returns>
+        public List<CachedDriveItem> FilterPlayableMedia(List<CachedDriveItem> unfilteredList)
+        {
+            if (unfilteredList is null)
+            {
+                return null;
+            }
+
+            List<CachedDriveItem> filteredList = new List<CachedDriveItem>();
+            foreach(CachedDriveItem currentItem in unfilteredList)
+            {
+                // Item is playable if not a folder but file
+                if (!currentItem.IsFolder)
+                {
+                    filteredList.Add(currentItem);
+                }
+            }
+            return filteredList;
+        }
+
```

Comment:
I don't know if it is the task of a CacheHelper to filter the items for the previous /next list. But i can't really think of another way. But it's maybe a good idea if we take a look at this when we redesign the structure of the application.

Reply:
Frankly, I'm not sure if it is the responsibility of the App to hold a list of media items either. But what can you do.

Reply:
You are right. But it's the only place at the moment we can use. To pass a list, we dont know the size of, to another page in a parameter with `navigation` seems to me it can cause trouble down the road.